### PR TITLE
Correctly 'load' minimized clients to a frame tree

### PR DESCRIPTION
**Pull request by t-wissmann**
This makes the 'load' command handle minimized clients correctly and
adds test cases for it. I'm also setting 'floating' to 'false', since
the client is sucked into the frame tree by the 'load' command and thus
definitely not in the floating state anymore.

### CODE REVIEW

File: src/frametree.cpp [hunk 1]
```diff
@@ -676,6 +676,10 @@ void FrameTree::applyFrameTree(shared_ptr<Frame> target,
         // this might even involve the above targetLeaf / targetSplit
         // so we need to do this before everything else
         for (const auto& client : sourceLeaf->clients) {
+            // first un-minimize and un-float the client
+            // such that we now that it is in the frame-tree
```
Comment (blarz) [inline src/frametree.cpp:680]: s/now/know/?

File: tests/test_load.py [hunk 2]
```diff
@@ -168,3 +168,46 @@ def test_fraction_precision_outside_range(hlwm):
...
             .expect_stderr('but actually is ' + v)
+
+
+def test_load_floating_client(hlwm):
+    winid, _ = hlwm.create_client()
+    hlwm.call(f'set_attr clients.{winid}.floating true')
+    hlwm.call('set_layout max')
+    assert hlwm.call('dump').stdout.rstrip() == '(clients max:0)'
+
+    # soak the client into the frame tree
```
Comment (blarz) [inline tests/test_load.py:179]: Isn't it "suck" more than "soak"?

File: tests/test_load.py [hunk 3]
```diff
@@ -168,3 +168,46 @@ def test_fraction_precision_outside_range(hlwm):
...
+    # and such that it is invisible initially
+    (True, True), (True, False), (False, True)
+])
+@pytest.mark.parametrize("floating", [True, False])
+def test_load_minimized_client(hlwm, othertag, minimized, floating):
+    if othertag:
+        hlwm.call('add othertag')
+        hlwm.call('rule tag=othertag')
+    winid, _ = hlwm.create_client()
+    if minimized:
```
Comment (blarz) [inline tests/test_load.py:199]: Could be removed

**Comment by t-wissmann** [comment]
Thanks, I've fixed all three issues.

File: tests/test_load.py (hunk 2 above)
Comment (blarz) [reply tests/test_load.py]: I also updated it in the commit message

**Review by blarz — APPROVED** [review]

**Comment by blarz** [comment]
> Thanks, I've fixed all three issues.

Thanks! Approved :)

**Pull request merged**